src/app/components/date-range-select/date-range-select.component.ts

```diff
@@ -0,0 +1,76 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { DateAdapter } from '@angular/material/core';
+import { Moment } from 'moment';
+import { DateSelectService } from 'src/app/services/date-select.service';
+
+@Component({
+  selector: 'app-date-range-select',
+  templateUrl: './date-range-select.component.html',
+  styleUrls: ['./date-range-select.component.scss']
+})
+export class DateRangeSelectComponent implements OnInit {
+  startDateCtrl = new FormControl(new Date());
+  endDateCtrl = new FormControl(new Date());
+  minDate = new Date(2020, 0, 1);
+  maxDate = new Date(Date.now());
+  isOpen = false;
+
+  constructor(private _adapter: DateAdapter<Moment>,private dateSelectService: DateSelectService) {
+    this._adapter.setLocale(navigator.language)
+    const { from, to } = dateSelectService.getDateRange();
+    this.startDateCtrl.setValue(from);
+    this.endDateCtrl.setValue(to);
+
+    // Ensures multiple date pickers are synced through the service
+    dateSelectService.getDateRangeObservable().subscribe(dateRange => {
+      this.startDateCtrl.setValue(dateRange.from);
+      this.endDateCtrl.setValue(dateRange.to);
+    })
+   }
+
+  ngOnInit(): void {
+  }
+
+  opened() {
+    this.isOpen = true;
+  }
+
+  closed() {
+    this.isOpen = false;
+  }
+
+  startDateChanged() {
+    if (this.validDatesAreSet() && !this.isOpen) { 
+      // need the isOpen tag or this would fire off a second event 
+      // when the date picker closes on selecting the end date
+      this.setNewRange();
+    }
+  }
```

Comment:
This was for the case where a user types in a date.
Only really came about because selecting ranges over multiple months are a bit annoying to do. Its easier just to edit the starting value by typing after entering a range if I want the same end date.

Reply:
OK, I didn't realise you can type in the date selector.